### DISCUSSION TITLE
libvips 8.13.3 generating bigger filesizes for PNG format than version 8.12.2

### DISCUSSION THREAD

**Discussion by johntrunc** [libvips/libvips#3156] [Q&A] [answered]
I have been testing PNG generation (size and performance) with libvips 8.13.3 and I have found that PNG files generated with this version of the library are much bigger than those generated with libvips 8.12.2.

In general I'm seeing an increase in the PNG filesizes of 30-50% compared to the same files generated with version 8.12.2 

At first I though the culprit was new libspng save generating bigger filesizes but after changing libvips to use only libpng the filesizes are the same (???).

Some info about my findings:

```sh
# vips-8.12.2 (using libpng 1.6.37)

$ vips pngsave example.jpeg example.png
# example.png, 55873745 bytes (55,9 MB)
$ vips pngsave example-02.jpeg example-02.png
# example-02.png, 10531717 bytes (10,5 MB)
$ vips pngsave example-03.jpeg example-03.png
# example-03.png, 12519983 bytes (12,5 MB)
```

```sh
# vips 8.13.3 (using libspng 0.7.2)

$ vips pngsave example.jpeg example.png
# example.png, 78579453 bytes (78,6 MB) (40.63% increase)
$ vips pngsave example-02.jpeg example-02.png
# example-02.png, 16348072 bytes (16,3 MB) (55.22% increase)
$ vips pngsave example-03.jpeg example-03.png
# example-03.png, 19427935 bytes (19,4 MB) (55.17% increase)
```

```sh
# vips 8.13.3 (using libpng 1.6.37)

$ vips pngsave example.jpeg example.png
# example.png, 78579453 bytes (78,6 MB) (40.63% increase)
$ vips pngsave example-02.jpeg example-02.png
# example-02.png, 16348072 bytes (16,3 MB) (55.22% increase)
$ vips pngsave example-03.jpeg example-03.png
# example-03.png, 19427935 bytes (19,4 MB) (55.17% increase)
```

You can download the used `example.jpeg` image [here](https://upload.wikimedia.org/wikipedia/commons/b/b7/The_sculptures_of_two_mythical_giant_demons%2C_Thotsakan_and_Sahatsadecha%2C_guarding_the_eastern_gate_of_the_main_chapel_of_Wat_Arun%2C_Bangkok.jpg) if needed. 

Because the difference is huge I wanted to check that I'm not missing something, maybe something changed between versions that it's causing this big difference in the final filesize?

**Comment by lovell** [comment, marked as answer]
Did you see the changelog?

The following entry might be relevant, especially as your examples appear to be re-encoding JPEG as PNG.

https://github.com/libvips/libvips/blob/e24cee4e224af32d85ab2fd52f57c2482cc87d9c/ChangeLog#L91

**Reply by johntrunc** [reply]
Thanks @lovell I totally miss that change!.
